stream-chat-android-offline/src/main/java/io/getstream/chat/android/offline/message/attachments/internal/UploadAttachmentsWorker.kt

```diff
@@ -26,11 +26,14 @@ import io.getstream.chat.android.client.utils.ProgressCallback
 import io.getstream.chat.android.client.utils.Result
 import io.getstream.chat.android.client.utils.SyncStatus
 import io.getstream.chat.android.client.utils.recover
-import io.getstream.chat.android.offline.plugin.logic.channel.internal.ChannelLogic
 import io.getstream.chat.android.offline.plugin.logic.internal.LogicRegistry
+import io.getstream.chat.android.offline.plugin.state.StateRegistry
+import io.getstream.chat.android.offline.plugin.state.channel.internal.ChannelMutableState
+import io.getstream.chat.android.offline.plugin.state.channel.internal.toMutableState
 
 internal class UploadAttachmentsWorker(
     private val logic: LogicRegistry,
+    private val state: StateRegistry,
```

Comment:
Hey @leandroBorgesFerreira,

I wonder if we can pass `ChannelMutableState` or at least a function that returns `ChannelMutableState` to avoid having `StateRegistry` here.

Reply:
Other than that it's looking good to me.

Reply:
Actually, I misread the code haha. I can pass the ChannelState.